packages/astro/src/core/dev/index.ts

```diff
@@ -73,7 +73,7 @@ export class AstroDevServer {
     this.port = config.devOptions.port;
     this.origin = `http://localhost:${this.port}`;
     this.site = config.buildOptions.site ? new URL(config.buildOptions.site) : undefined;
-    this.pathname = this.site ? this.site.pathname + '/' : '/';
+    this.pathname = this.site ? this.site.pathname : '/';
```

Comment:
This was forcing to always end with `/` causing some `//` when including that in the `buildOptions.site` config.